modules/guardian/genesis.go

```diff
@@ -6,11 +6,11 @@ import (
 
 // GenesisState - all guardian state that must be provided at genesis
 type GenesisState struct {
-	Profilers []Profiler `json:"profilers"`
-	Trustees  []Trustee  `json:"trustees"`
+	Profilers []Guardian `json:"profilers"`
+	Trustees  []Guardian `json:"trustees"`
 }
 
-func NewGenesisState(profilers []Profiler, trustees []Trustee) GenesisState {
+func NewGenesisState(profilers, trustees []Guardian) GenesisState {
```

Comment:
At genesis, we should ensure Guardian.AddressType == genesis

Reply:
Yes, I will add AddressType check.

Reply:
We should add a new function to check GenesisState.